src/utils/renderHelpers.ts

```diff
@@ -15,6 +15,28 @@ export interface NodeRenderOptions {
 	start?: number;
 	end?: number;
 	isNoStatement?: boolean;
+	renderedParent?: Node;
+	fieldOfRenderedParent?: string | null;
+}
+
+export function getFieldOfParent(child: Node): string | null {
+	const parent = <Node>child.parent;
+	for (const key of parent.keys) {
+		const value = (<any>parent)[key];
+		if (value === child) return key;
+		if (Array.isArray(value)) {
+			for (const nestedValue of value) {
+				if (nestedValue === child) return key;
+			}
+		}
+	}
+	return null;
+}
+
+export function childIsStatement(parent: { type?: string }) {
+	return (
+		parent.type === 'Program' || parent.type === 'ExpressionStatement' // e.g. default exports rendered for side-effects only
```

Comment:
I take it these will expand over time?

An alternative here might be `Program.prototype.isStatementContainer = true` and `ExpressionStatement.prototype.isStatementContainer = true` where `NodeBase.prototype.isStatementContainer = false`.

Reply:
Actually not. In fact at the moment if an expression is simplified so that it becomes a statement, there will always be an expression statement left as the actual parent *except* for simplified default exports. Changed this now to get rid of this exception.